libraries/aws/provisioning/src/aws_iot_provisioning_serializer.c

```diff
@@ -355,30 +377,173 @@ AwsIotProvisioningError_t _AwsIotProvisioning_SerializeCreateKeysAndCertificateR
 
 /*------------------------------------------------------------------*/
 
+AwsIotProvisioningError_t _serializeCertFromCsrPayload( const char * pCertificateSigningRequest,
+                                                        size_t csrLength,
+                                                        IotSerializerEncoderObject_t * pEncoder,
+                                                        bool isDrySerialization )
+{
+    AwsIotProvisioningError_t status = AWS_IOT_PROVISIONING_SUCCESS;
+
+    IotSerializerEncoderObject_t mapEncoder = IOT_SERIALIZER_ENCODER_CONTAINER_INITIALIZER_MAP;
+    IotSerializerError_t serializerResult = IOT_SERIALIZER_SUCCESS;
+    IotSerializerScalarData_t csrData = IotSerializer_ScalarTextStringWithLength( pCertificateSigningRequest,
+                                                                                  csrLength );
+
+    /* Determine the status checking expression logic for the serializer returned
+     * status code based on whether the we are performing a dry-serialization or not.
+     */
+    bool (* isSuccessStatus)( IotSerializerError_t );
+
+    if( isDrySerialization == true )
+    {
+        isSuccessStatus = _checkSuccessOrBufferToSmall;
+    }
+    else
+    {
+        isSuccessStatus = _checkSuccess;
+    }
+
+    /* Encode the payload as a map container. */
+    serializerResult = _pAwsIotProvisioningEncoder->openContainer( pEncoder,
+                                                                   &mapEncoder,
+                                                                   1 );
```

Comment:
What does `1` mean? Also, what file is `IotSerializerEncoderObject_t` defined in? I wasn't able to find it by searching on GitHub

Reply:
`1` is the number of entries in the CBOR map. I can replace with a macro for better expressivity.

Reply:
Yes, I think that's appropriate

Reply:
Pushed